sljit_src/sljitNativeX86_64.c

```diff
@@ -805,7 +805,7 @@ static sljit_s32 call_with_args(struct sljit_compiler *compiler, sljit_s32 arg_t
 	if (word_arg_count == 0)
 		return SLJIT_SUCCESS;
 
-	if (word_arg_count >= 3) {
+	if (word_arg_count >= 1) {
```

Comment:
Shouldn't it be: `(word_arg_count >= 3 || src == SLJIT_R2)`. Perhaps an `xchg` could be used to use a single mov.
Could you add a test as well?